plugins/by-name/friendly-snippets/default.nix

```diff
@@ -1,31 +1,13 @@
 {
   lib,
-  helpers,
-  config,
-  pkgs,
   ...
 }:
-with lib;
-let
-  cfg = config.plugins.friendly-snippets;
-in
-{
-  meta.maintainers = [ maintainers.GaetanLepage ];
-
-  options.plugins.friendly-snippets = {
-    enable = mkEnableOption "friendly-snippets";
-
-    package = lib.mkPackageOption pkgs "friendly-snippets" {
-      default = [
-        "vimPlugins"
-        "friendly-snippets"
-      ];
-    };
-  };
+lib.nixvim.vim-plugin.mkVimPlugin {
```

Comment:
This is a lua plugin, though https://github.com/rafamadriz/friendly-snippets

Reply:
Yes, but as it is not configured using lua code, we don't consider it as a "neovim" plugin.
Plugins that have no configuration are implemented using `mkVimPlugin`.

Maybe long term we can find a better abstraction. In the short-term we'll stick with this methodology.

Reply:
We've been using `hasSettings = false` and `callSetup = false`

Reply:
Sure, but this makes not much more sense to me...
The `mkNeovimPlugin` name is a bit misleading, it should be `mkPluginThatIsConfiguredWithRequireLuaOrSomethingSimilar`.

I am open to discussing how to rethink those helpers and their respective field.
However, using `mkVimPlugin` here is fine in my opinion.

Reply:
> We've been using `hasLuaConfig = false`, `hasSettings = false`, and `callSetup = false`

The fact we can now use either `mkNeovimPlugin` or `mkVimPlugin` makes me wonder about combining them together into on single function?

Or alternatively, maybe a lot of the modes available via args should actually be special-case wrapper functions that wrap a single lower-level `mkPluginWith`?